crates/cli/src/templates/binding.java

```diff
@@ -0,0 +1,62 @@
+package PARSER_NS_CLEANED.jtreesitter.LOWER_PARSER_NAME;
+
+import java.lang.foreign.*;
+
+public final class PARSER_CLASS_NAME {
+    private static final ValueLayout VOID_PTR =
+            ValueLayout.ADDRESS.withTargetLayout(MemoryLayout.sequenceLayout(Long.MAX_VALUE, ValueLayout.JAVA_BYTE));
+    private static final FunctionDescriptor FUNC_DESC = FunctionDescriptor.of(VOID_PTR);
+    private static final Linker LINKER = Linker.nativeLinker();
+    private static final TreeSitterJava INSTANCE = new PARSER_CLASS_NAME();
+
+    private final Arena arena = Arena.ofAuto();
+    private SymbolLookup lookup = null;
```

Comment:
Maybe the `lookup` field should be `volatile`; otherwise you could get difficult to debug issues when `language()` is called by separate threads (not necessarily concurrently).

Using `volatile` solves some of the thread-safety issues but you could still end up with a race-condition where two threads use different `lookup` values; not sure if that is safe. If not might even need synchronization.

See related https://wiki.sei.cmu.edu/confluence/display/java/LCK10-J.+Use+a+correct+form+of+the+double-checked+locking+idiom

Reply:
> you could still end up with a race-condition where two threads use different `lookup` values

We can deal with that if and when it happens.